pkg/dataingestionframework/data/dif_types.go

```diff
@@ -0,0 +1,52 @@
+package data
+
+import "fmt"
+
+//Data ingestion framework topology entity
+type DIFEntity struct {
+	UID  string `json:"uniqueId"`
+	Type string `json:"type"`
+	Name string `json:"name"`
+
+	HostedOn            *DIFHostedOn                 `json:"hostedOn"`
+	MatchingIdentifiers *DIFMatchingIdentifiers      `json:"matchIdentifiers"`
+	PartOf              []*DIFPartOf                 `json:"partOf"`
+	Metrics             []map[string][]*DIFMetricVal `json:"metrics"`
```

Comment:
The `Metrics` field seems rather complicated to me, why can't we define the following:
```
Metrics             map[string]*DIFMetricVal `json:"metrics"`
```
where the key is the metric type, and the value is the `DIFMetricVal` structure which has the values of different kind such as average and capacity?